src/dataService/tagDataService.ts

```diff
@@ -182,31 +186,42 @@ export class TagDataService {
    * is actually in the user's list (no foreign key constraint).
    * @param tagInputs
    */
-  public async insertTags(tagInputs: TagCreateInput[]): Promise<void> {
+  public async insertTags(tagInputs: TagSaveAssociation[]): Promise<void> {
     await this.db.transaction(async (trx: Knex.Transaction) => {
       await this.insertTagAndUpdateSavedItem(tagInputs, trx);
       await this.usersMetaService.logTagMutation(new Date(), trx);
     });
   }
 
   private async insertTagAndUpdateSavedItem(
-    tagInputs: TagCreateInput[],
-    trx: Knex.Transaction<any, any[]>
+    tagInputs: TagSaveAssociation[],
+    trx: Knex.Transaction<any, any[]>,
+    timestamp?: Date
   ) {
-    const timestamp = mysqlTimeString(new Date(), config.database.tz);
+    if (tagInputs.length === 0) {
```

Comment:
wouldn't this be caught by constraints? but i guess they're commented out atm

Reply:
yeah, more a caution since this is a private function that isn't specifically tied to that resolver